android/src/main/kotlin/co/doneservices/callkeep/CallKeepNotificationManager.kt

```diff
@@ -0,0 +1,439 @@
+        notificationBuilder.setSubText(if (TextUtils.isEmpty(missedCallText)) context.getString(R.string.text_missed_call) else missedCallText)
+    
+        val contentTitle = data.getString(EXTRA_CALLKEEP_CONTENT_TITLE, "")
+        if(contentTitle?.isNotEmpty() == true){
+            notificationBuilder.setContentTitle(contentTitle)
+        } else {
+            notificationBuilder.setContentTitle(data.getString(EXTRA_CALLKEEP_CALLER_NAME, ""))
+        }
+        notificationBuilder.setContentText(data.getString(EXTRA_CALLKEEP_HANDLE, ""))
+        val avatarUrl = data.getString(EXTRA_CALLKEEP_AVATAR, "")
+        if (avatarUrl != null && avatarUrl.isNotEmpty()) {
+            val headers =
+                    data.getSerializable(CallKeepBroadcastReceiver.EXTRA_CALLKEEP_HEADERS) as HashMap<String, Any?>
+
+            getPicassoInstance(context, headers).load(avatarUrl)
+                    .into(targetLoadAvatarDefault)
+        }
+        val showCallBackAction = data.getBoolean(CallKeepBroadcastReceiver.EXTRA_CALLKEEP_SHOW_CALLBACK, true)
+        if (showCallBackAction) {
+            val callBackText = data.getString(EXTRA_CALLKEEP_CALLBACK_TEXT, "")
+            val callbackAction: NotificationCompat.Action = NotificationCompat.Action.Builder(
+                    R.drawable.ic_accept,
+                    if (TextUtils.isEmpty(callBackText)) context.getString(R.string.text_call_back) else callBackText,
+                    getCallbackPendingIntent(notificationId, data)
+            ).build()
+            notificationBuilder.addAction(callbackAction)
+        }
+        
+        notificationBuilder.priority = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            NotificationManager.IMPORTANCE_HIGH
+        } else {
+            Notification.PRIORITY_HIGH
+        }
+        notificationBuilder.setSound(missedCallSound)
+        notificationBuilder.setContentIntent(getAppPendingIntent(notificationId, data))
+        val accentColor = data.getString(EXTRA_CALLKEEP_ACCENT_COLOR, "#4CAF50")
+        try {
+            notificationBuilder.color = Color.parseColor(accentColor)
+        } catch (error: Exception) {
+        }
+
+        val notification = notificationBuilder.build()
+        getNotificationManager().notify(notificationId, notification)
+        Handler(Looper.getMainLooper()).postDelayed({
+            try {
+                getNotificationManager().notify(notificationId, notification)
+            } catch (error: Exception) {
+            }
+        }, 1000)
+    }
+
+
+    fun clearIncomingNotification(data: Bundle) {
+        context.sendBroadcast(IncomingCallActivity.getIntentEnded(context))
+        notificationId = data.getString(EXTRA_CALLKEEP_ID, "callkeep").hashCode()
+        getNotificationManager().cancel(notificationId)
+    }
+
+    fun clearMissCallNotification(data: Bundle) {
+        notificationId = data.getString(EXTRA_CALLKEEP_ID, "callkeep").hashCode()
```

Comment:
This has the same notification id as above?

Reply:
This is notification id, missed cal notification only shows up after the incoming call notification is dismissed through timeout (we don't use it in our app though as we send our own).
I think it's easier and more consistent to reference notifications by the call's UUID as it's already unique.